virttest/vt_utils/cpu.py

```diff
@@ -0,0 +1,103 @@
+#
+# library for cpu related helper functions
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; specifically version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat (c) 2023
+# Author: Houqi Zuo <hzuo@redhat.com>
+import re
+
+from avocado.utils import process
+
+
+def get_cpu_info():
+    """
+    Return information about the CPU information.
+
+    :return: cpu information
+    :rtype: dict
+    """
+    cpu_info = dict()
+    with open("/proc/cpuinfo") as fd:
+        cpu_info_lines = fd.readlines()
+    for line in cpu_info_lines:
+        if line == "\n":
+            break
```

Comment:
hi @nickzhq , I am sorry that I may make a wrong comment about cpu info in the afternoon. For this function, we may return a list of dicts, in which each dict contains the information of one processor. Why? Because these information may be used for some purposes, e.g. to get the count of sockets, with shell, we can do it in this way: grep 'physical id' /proc/cpuinfo | sort | uniq | wc -l, i.e. we need all cpus' information.

or we can depend on lscpu to get the count of sockets. cc @YongxueHong